Assignment 3: Student Performance Analyzer

Tanishka33 opened the pull request:
Calculated the student performance , created different functions for reading students , calculating performance and displaying the data . Also added recursion for printing roll number .

Tanishka33 committed:
Completed Assignment 3

Tanishka33 committed:
Completed Assignment 3

Tanishka33 committed:
Removed .exe file

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+
+int main()
+{
+    int n;
```
use meaning full variable names , also do for rest

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+        printf("Enter the detail of students \n", &student[index]);
+
+        printf("Roll Number :");
+        scanf("%d", &student[index].rollNumber);
```
better to not print extra lines like "Enter Rollno." Be specific according to the assignment file output (output of yours should be same as pdf output not more than that)

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+
+        printf("Marks 3 : ");
+        scanf("%d", &student[index].marks[2]);
+    }
```
you can input these all in one line like 
scanf("%d %s %d %d %d", &students[index]->roll, students[index]->name, &students[index]->marks[0], &students[index]->marks[1], &students[index]->marks[2]);

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+    printf("\n");
+
+    return 0;
+}
```
add EndLine

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+        {
+        case 'A':
+            printf("Performance: *****");
+            break;
```
Can use this in diff function would better and clean looking

vidhyang789 commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,138 @@
+        case 'C':
+            printf("Performance: ***");
+            break;
+        case 'D':
+            printf("Performance: **");
+            break;
```
also using printf multiple times is not good rather can use a variable and then print in last

vidhyang789 reviewed (commented):
change the things commented below and add the last empty line too

Tanishka33 committed:
Made the changes

Tanishka33 commented:
Made all changes: updated variable names, removed extra prints, used one-line input, and better performance printing.
but, 
Skipped using a variable for stars , as it will add unnecessary complexity without improving readability or performance.

Tanishka33 committed:
Added for loop for printing stars

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,150 @@
+#include <math.h>
+#include <ctype.h>
+
+typedef struct StudentPerformanceAnalyzer
```
Keep it as Student

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,150 @@
+void printRollNumber(studentPerformance student[], int index, int studentCount);
+void printPerformance(char grade);
+
+void inputDetails(studentPerformance student[], int studentCount)
```
use const in studentCount, takecare in other places where you can use const.

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,150 @@
+            student[index].grade = 'D';
+        }
+
+        else if (student[index].average < 35)
```
this is else

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,150 @@
+        return;
+    }
+    printf("%d ", student[index].rollNumber);
+
```
extra line

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <ctype.h>
```
is this header are required in this program.

Tanishka33 replied:
I added those header files out of habit .. realized those are not needed , now i have cleaned those .. only kept important <stdio.h>

Tanishka33 committed:
Renamed struct, added const to parameters, simplified else, and removed unused headers.

Tanishka33 commented:
Renamed struct, added const to parameters, and simplified else, and removed unused headers.

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,143 @@
+               student[index].average,
+               student[index].grade);
+
+        if(student[index].grade != 'F')
```
use continue statement to meet the assignment requirement. format.

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,143 @@
+{
+    int starCount = 0;
+    switch (grade)
+        {
+        case 'A':
+            starCount = 5;
+            break;
+        case 'B':
+            starCount = 4;
+            break;
+        case 'C':
+            starCount = 3;
+            break;
+        case 'D':
+            starCount = 2;
+            break;
+        case 'F':
+            break;
+        default:
+            break;
+        }
+        
+        for(int index = 0; index < starCount; index++)
+        {
+            printf("*");
```
format

tsraviteja reviewed (commented):
Look's good, minor comments are given.

Tanishka33 commented:
can I resolve the previous comments ??

tsraviteja commented:
> can I resolve the previous comments ??

Yes

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,143 @@
+
+void inputDetails(Student student[], const int studentCount);
+void calculateResults(Student student[], const int studentCount);
+void displayResult(Student student[], const int studentCount);
```
meaningful method names

Tanishka33 committed:
declared all functions with meaningfull names

Tanishka33 commented:
declared all functions with meaningfull names

tsraviteja commented on Assignment-3/studentPerformance.c:
```diff
@@ -0,0 +1,143 @@
+        if(student[index].grade != 'F')
+        {
+            printf("Performance: ");
+            printPerformanceStars(student[index].grade);
```
check there is compilation issuse.

Tanishka33 replied:
Sorry there was mismatch in function declaration , I have fixed it now .. its working properly

Tanishka33 committed:
Sorry, there was mismatch in function declaration, fixed it!
